src/ModelObjects/representations.rs

```diff
@@ -60,6 +137,32 @@ pub enum SystemRepresentation {
 }
 
 impl<'a> SystemRepresentation {
+    pub fn get_max_bounds(&self, dimensions: u32) -> MaxBounds {
+        let mut bounds = MaxBounds::create(dimensions);
+
+        match self {
+            SystemRepresentation::Composition(left_side, right_side) => {
+                bounds.add_bounds(&mut left_side.get_max_bounds(dimensions));
+                bounds.add_bounds(&mut right_side.get_max_bounds(dimensions));
+            }
+            SystemRepresentation::Conjunction(left_side, right_side) => {
+                bounds.add_bounds(&mut left_side.get_max_bounds(dimensions));
+                bounds.add_bounds(&mut right_side.get_max_bounds(dimensions));
+            }
+            SystemRepresentation::Parentheses(rep) => {
+                bounds.add_bounds(&mut rep.get_max_bounds(dimensions))
+            }
+            SystemRepresentation::Component(comp) => {
+                let mut comp_bounds = comp.get_max_bounds(dimensions);
+                //Im not sure why this transformation is needed
```

Comment:
I'm not sure either, what happens if we do not do the transformation? it's not really a problem that we have to do the transformation as 1 > 0 but it is still strange.